js/header-resizer.js

```diff
@@ -0,0 +1,63 @@
+/* global $ $$ dom */// dom.js
+/* global debounce */// toolbox.js
+/* global prefs */
+'use strict';
+
+(() => {
+  let curW, offset, active;
+  prefs.subscribe(dom.HWprefId, (key, val) => {
+    if (!active && val !== curW) {
+      getCurWidth();
+      setWidth(val);
+    }
+  });
+  $('#header-resizer').onmousedown = e => {
+    if (e.button) return;
+    getCurWidth();
+    offset = curW - e.clientX;
+    active = true;
+    document.body.classList.add('resizing-h');
+    document.on('mousemove', resize);
+    document.on('mouseup', resizeStop);
+  };
+
+  function getCurWidth() {
+    curW = parseFloat(document.documentElement.style.getPropertyValue('--header-width'))
+      || $('#header').offsetWidth;
+  }
+
+  /** @param {MouseEvent} e */
+  function resize(e) {
+    if (setWidth(offset + e.clientX)) {
+      debounce(save, 250, e.shiftKey);
```

Comment:
We used to update prefs at real-time. Now it only updates prefs when the mouse stopped. Is this intended?

Reply:
If we don't want to save at real-time, it might be better to save on mouseup. Otherwise, users may get some confusing results if they tap the shift key while dragging.

Or, use the shift key from mousedown event.